workflow/controller/workflowpod.go

```diff
@@ -690,17 +690,8 @@ func (woc *wfOperationCtx) newExecContainer(name string, tmpl *wfv1.Template) *a
 		Image:           woc.controller.executorImage(),
 		ImagePullPolicy: woc.controller.executorImagePullPolicy(),
 		Env:             woc.createEnvVars(),
-	}
-	if woc.controller.Config.Executor != nil {
-		exec.Args = woc.controller.Config.Executor.Args
```

Comment:
Seems like this got unintentionally removed -- see https://github.com/argoproj/argo-workflows/issues/12315#issuecomment-1913359846